### PR TITLE
second-order BK failure patch

### PR DESCRIPTION
**Pull request by wupeifan**
This PR fixes the issue that when BK fails, the second-order perturbation solution triggers an error instead of returning an error message. I added a corresponding unit test as well.

### CODE REVIEW

**Comment by wupeifan** [comment]
CI passes.

File: Project.toml [hunk 1]
```diff
@@ -29,7 +28,6 @@ Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f"
 
 [compat]
 ChainRulesCore = "1"
```
Comment (cpfiffer) [inline Project.toml:30]:
```suggestion
ChainRulesCore = "1"
DifferenceEquations="0.2"
```

File: Project.toml [hunk 2]
```diff
@@ -5,7 +5,6 @@ version = "0.4.2"
 
 [deps]
 ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
```
Comment (cpfiffer) [inline Project.toml:7]:
```suggestion
ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
DifferenceEquations = "e0ca9c66-1f9e-11ec-127a-1304ce62169c"
```

**Review by cpfiffer — APPROVED** [review]
Love it, let's put it in. I added two suggestions to make sure `DifferenceEquations` stays in, but these will break CI so we are alright to merge this in without tests if those are committed.

**Comment by wupeifan** [comment]
I still don't understand why we need DifferenceEquations. Maybe discuss on slack?

**Pull request merged**